### PR TITLE
add readwise-reader 0.1.1 (new cask)

### PR DESCRIPTION
**Pull request by 0xdevalias**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

This app is an alpha release because it only launched today, but it will become the canonical stable release 'soon'.

---

This cask is named in line with the brand name used for the equivalent app on iOS:

- https://apps.apple.com/us/app/readwise-reader/id1567599761

And the name they use for the section on the website:

- https://readwise.io/read
  - image

'Readwise' is both the name of the company, and of another older product they have.

An existing Readwise app's Cask follows a similar naming pattern here:

- https://github.com/Homebrew/homebrew-cask/blob/master/Casks/r/readwise-ibooks.rb#L5

---

I attempted to figure out how to add livecheck info, but wasn't able to figure it out:

```
⇒ $(brew --repository homebrew/cask)/developer/bin/find-appcast /Applications/Reader.app

Looking for Sparkle appcast… Not found.
Looking for electron-builder appcast… Not found.
```

### CODE REVIEW

**Comment by 0xdevalias** [comment]
Running with Proxyman, it seems to check the following URL for updates:

- https://reader-desktop-releases.readwise.io/update-manifest.json

Resulting in json like this:

```
{
  "version": "0.1.1",
  "notes": "Readwise Reader",
  "pub_date": "2023-12-12T20:26:12.298772Z",
  "platforms": {
    "darwin-aarch64": {
      "signature": "dW50cnVzdGVkIGNvbW1lbnQ6IHNpZ25hdHVyZSBmcm9tIHRhdXJpIHNlY3JldCBrZXkKUlVUdDd4Ukw4YUhFb2V1NFF1a3FhN2dnNlhLOWVSVXlSNmx3bU56d2V5T1dQbW11bTNZdm5wU0hVOC9FNjg3NE81V2VudUlmV0NqV2xPN3VpdlRYK3RRTk9tK1FoaEw0WHdJPQp0cnVzdGVkIGNvbW1lbnQ6IHRpbWVzdGFtcDoxNzAyNDMwNzIzCWZpbGU6UmVhZGVyLmFwcC50YXIuZ3oKaWVKZHNyRVJ6eTAyMVZ3azFpdFJPUkwrb0syRU0wQWJJT0h3MUpadTlKY0g3Nklva2J4RWNmRnZOMk5OSk5JUXYxcUhWUTFHZ3dLTmRBUzBmakY2RGc9PQo=",
      "url": "https://reader-desktop-releases.readwise.io/versions/0.1.1/macos/Reader.app.tar.gz"
    },
    "darwin-x86_64": {
      "signature": "dW50cnVzdGVkIGNvbW1lbnQ6IHNpZ25hdHVyZSBmcm9tIHRhdXJpIHNlY3JldCBrZXkKUlVUdDd4Ukw4YUhFb2V1NFF1a3FhN2dnNlhLOWVSVXlSNmx3bU56d2V5T1dQbW11bTNZdm5wU0hVOC9FNjg3NE81V2VudUlmV0NqV2xPN3VpdlRYK3RRTk9tK1FoaEw0WHdJPQp0cnVzdGVkIGNvbW1lbnQ6IHRpbWVzdGFtcDoxNzAyNDMwNzIzCWZpbGU6UmVhZGVyLmFwcC50YXIuZ3oKaWVKZHNyRVJ6eTAyMVZ3azFpdFJPUkwrb0syRU0wQWJJT0h3MUpadTlKY0g3Nklva2J4RWNmRnZOMk5OSk5JUXYxcUhWUTFHZ3dLTmRBUzBmakY2RGc9PQo=",
      "url": "https://reader-desktop-releases.readwise.io/versions/0.1.1/macos/Reader.app.tar.gz"
    }
  }
}
```

File: Casks/r/readwise-reader.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cask "readwise-reader" do
```
Comment (bevanjkay) [inline Casks/r/readwise-reader.rb:1]: I think the `token` here should just be `reader`, as we essentially always follow the `app` stanza in line with the [token reference documentation](https://docs.brew.sh/Cask-Cookbook#token-reference).

In the case of your example `readwise-ibooks`, this also follows the `app` stanza.

File: Casks/r/readwise-reader.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  sha256 "f8328a701bdefbd839c484c57827bd05b56be4216722dfe72c4348b823f9048d"
+
+  url "https://reader-desktop-releases.readwise.io/versions/#{version}/macos/Reader.app.tar.gz"
+  name "Readwise Reader"
+  desc "Save articles to read, highlight key content, and organize notes for review"
+  homepage "https://readwise.io/read/"
+
+  livecheck do
+    url "https://reader-desktop-releases.readwise.io/update-manifest.json"
+    regex(/"version":\s*"([^"]+)"/i)
```
Comment (bevanjkay) [inline Casks/r/readwise-reader.rb:12]:
```suggestion
    strategy :json do |json|
      json["version"]
    end
```

File: Casks/r/readwise-reader.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+cask "readwise-reader" do
+  version "0.1.1"
+  sha256 "f8328a701bdefbd839c484c57827bd05b56be4216722dfe72c4348b823f9048d"
+
+  url "https://reader-desktop-releases.readwise.io/versions/#{version}/macos/Reader.app.tar.gz"
+  name "Readwise Reader"
+  desc "Save articles to read, highlight key content, and organize notes for review"
```
Comment (bevanjkay) [inline Casks/r/readwise-reader.rb:7]: I think the [`desc`](https://docs.brew.sh/Cask-Cookbook#stanza-desc) stanza could be a bit clearer, but I'm struggling to find a good way to describe the app!

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/r/readwise-reader.rb (hunk 1 above)
Comment (0xdevalias) [reply Casks/r/readwise-reader.rb:1]: I would say these parts apply:

> If the result of that process is a generic term, such as “Macintosh Installer”, try prepending the name of the vendor or developer, followed by a hyphen. If that doesn’t work, then just create the best name you can, based on the vendor’s web page.
> 
> If the result conflicts with the name of an existing cask, make yours unique by prepending the name of the vendor or developer, followed by a hyphen. Example: [unison.rb](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/u/unison.rb) and [panic-unison.rb](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/p/panic-unison.rb).

Pretty much everything else 'Reader' has a vendor prefix on it:

```
⇒ brew search reader
==> Formulae
readpe                          reaver                          trader                          redir                           redex

==> Casks
abbyy-finereader-pdf            fbreader                        html-mangareader                quiet-reader                    renamer
adobe-acrobat-reader            fluent-reader                   koodo-reader                    radar                           tableau-reader
cisdem-document-reader          font-newsreader                 neat-reader                     raven-reader                    yacreader
ebibookreader                   foxitreader                     pdf-reader-pro                  reaper
```

File: Casks/r/readwise-reader.rb (hunk 3 above)
Comment (0xdevalias) [reply Casks/r/readwise-reader.rb:7]: Yeah, I struggled as well. All the stuff they have written on the site/etc feels too pointless/marketing'ish/etc.

File: Casks/r/readwise-reader.rb (hunk 1 above)
Comment (bevanjkay) [reply Casks/r/readwise-reader.rb:1]: None of these are actually "vendor prefixes" the `app` for each of these is actually named how the token is. 

```
brew info fluent-reader neat-reader koodo-reader foxitreader raven-reader tableau-reader
==> fluent-reader: 1.1.4
https://hyliu.me/fluent-reader/
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Fluent Reader
==> Description
RSS/Atom news aggregator
==> Artifacts
Fluent Reader.app (App)
==> Analytics
install: 23 (30 days), 64 (90 days), 197 (365 days)

==> neat-reader: 9.0.10
https://www.neat-reader.com/
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Neat Reader
==> Description
Read, annotate and manage ePub books
==> Artifacts
NeatReader.app (App)
==> Analytics
install: 12 (30 days), 41 (90 days), 104 (365 days)

==> koodo-reader: 1.5.1
https://koodo.960960.xyz/en
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Koodo Reader
==> Description
Open-source epub reader
==> Artifacts
Koodo Reader.app (App)
==> Analytics
install: 82 (30 days), 284 (90 days), 696 (365 days)

==> foxitreader: 2023.3
https://www.foxit.com/pdf-reader/
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Foxit Reader
==> Description
PDF reader
==> Artifacts
FoxitPDFReader20233.L10N.Setup.pkg (Pkg)
==> Analytics
install: 116 (30 days), 405 (90 days), 1,023 (365 days)

==> raven-reader: 1.0.80 (auto_updates)
https://ravenreader.app/
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Raven Reader
==> Description
News reader with flexible settings
==> Artifacts
Raven Reader.app (App)
==> Analytics
install: 8 (30 days), 28 (90 days), 120 (365 days)

==> tableau-reader: 2023.3.0
https://www.tableau.com/products/reader
Not installed
From: git@github.com:Homebrew/homebrew-cask.git
==> Name
Tableau Reader
==> Description
Open and interact with data visualizations built in Tableau Desktop
==> Artifacts
Tableau Reader.pkg (Pkg)
==> Analytics
install: 6 (30 days), 12 (90 days), 131 (365 days)
```

Comment (0xdevalias) [reply Casks/r/readwise-reader.rb:1]: It still feels like "Reader" on it's own is a super generic and inaccurate name when the app is "Readwise Reader" 🤷🏻

Comment (bevanjkay) [reply Casks/r/readwise-reader.rb:1]: I can see that Readwise often refer to it as just "Reader". What is the name in the menu bar when the app is open?

Comment (bevanjkay) [reply Casks/r/readwise-reader.rb:1]: To be clear we don't apply the exceptions very often, if ever, except for cases where the token is already taken, or the token would result in the cask being mistaken as an official app for a service/product when it isn't. 

The guidelines help us to make quick decisions and maintain consistency across many submissions and casks.

Comment (0xdevalias) [reply Casks/r/readwise-reader.rb:1]: > What is the name in the menu bar when the app is open?

The apple menu bar is 'Reader':

image

The application window is 'Readwise Reader':

image

The iOS app is 'Readwise Reader' on the app store:

- https://apps.apple.com/us/app/readwise-reader/id1567599761

image

But 'Reader' on device:

IMG_0156

In my experience using their web/mobiles apps, they are pretty terrible at brand consistency.

---

> To be clear we don't apply the exceptions very often, if ever, except for cases where the token is already taken, or the token would result in the cask being mistaken as an official app for a service/product when it isn't.

@bevanjkay Fair. At the end of the day, 'Readwise Reader' feels like the more accurate brand; but ultimately I don't really care so much what the token ends up as, so long as it's 'discoverable' (by which I mean "when I know what I am looking for and try to find it, I can actually find it"), which it somewhat won't be until this is resolved:

- https://github.com/Homebrew/brew/issues/16237

**Comment by bevanjkay** [comment]
@0xdevalias Based on the `livecheck`, I would imagine this app has an auto-updating mechanism? If so, can you add `auto_updates true`.

Once that and the `token` is updated, we should be good to merge here.

**Comment by 0xdevalias** [comment]
> @Based on the `livecheck`, I would imagine this app has an auto-updating mechanism? If so, can you add `auto_updates true`.

@bevanjkay That would seem to be a reasonable assumption; but not one I know how to validate currently since it's so new; without attempting to reverse engineer deeper aspects of the app to figure it out.

I'll check again, but I don't think I even saw a 'check for updates' button; so was a little surprised when that URL popped up in the proxy.

> Once that and the `token` are updated, we should be good to merge here.

@bevanjkay Sounds good

**Comment by krehel** [comment]
Late to the party with a potentially unpopular opinion @bevanjkay - personally I'm not sure on calling it just `reader`, on grounds it's not really a distinguishable / identifiable name.

Yeah, it's anti-guideline, but I think `readwise-reader` is more... I'm not sure. Specific?

No conflict meant! 💛

**Comment by bevanjkay** [comment]
> personally I'm not sure on calling it just `reader`, on grounds it's not really a distinguishable / identifiable name.

I think the same could be said for many other tokens, `console`, `craft` etc... 
We haven't had an app submitted with the token `reader` in all the years `homebrew-cask` has been a thing.
The name "Reader" without the vendor name seems to be pretty consistent through the developer's branding.

I err on the side of sticking with consistency and not creating exceptions (as it creates a precedence to do the same in other occurrences), and this would be my preference. But I'm not going to split hairs over it.

**Comment by krehel** [comment]
Extremely fair points, especially regarding history and not seeing this before.

Sold 👍 I retract my statements.🙂

**Comment by 0xdevalias** [comment]
> I err on the side of sticking with consistency and not creating exceptions (as it creates a precedence to do the same in other occurrences), and this would be my preference.

@bevanjkay  Another alternative (that is completely OOS of this issue, and I probably don't care enough to raise it wider at this stage.. but one day I might, and then I can reference back to this) but I wonder if those guidelines should be updated to reference the common brand/'well known' name of the app; rather than being so heavily focussed on the name of the `*.app`/`*.pkg` name.

That said, I suspect this was chosen as the starting point as it's easier to automate; and I assume the change I proposed here would potentially lead to more  'discussions' over the nuance of what the 'well known' brand of things is. Though given I've raised that on a number of Casks I've been involved in recently; perhaps that discussion happens anyway, so refining the naming guidelines wouldn't dramatically change that. (And potentially most people don't have as strongly held feelings about the name used for the token as much as I seem to :p )

**Comment by bevanjkay** [comment]
> Though given I've raised that on a number of Casks I've been involved in recently; perhaps that discussion happens anyway, so refining the naming guidelines

I would say that in the two and a half years that I have been a maintainer for `homebrew-cask` we have had a discussion about the specific naming of a cask 2-3 times. Usually the `generate-cask-token` script is ran, and that token is used.

**Comment by 0xdevalias** [comment]
> I would say that in the two and a half years that I have been a maintainer for homebrew-cask we have had a discussion about the specific naming of a cask 2-3 times

@bevanjkay _counts the number of times I have started that discussion recently_

Perhaps it's just me then :p

**Comment by 0xdevalias** [comment]
> > Based on the `livecheck`, I would imagine this app has an auto-updating mechanism? If so, can you add `auto_updates true`.
> 
> @bevanjkay That would seem to be a reasonable assumption; but not one I know how to validate currently since it's so new; without attempting to reverse engineer deeper aspects of the app to figure it out.
> 
> I'll check again, but I don't think I even saw a 'check for updates' button; so was a little surprised when that URL popped up in the proxy.

@bevanjkay Checked again, couldn't find a 'check for updates' button; tried intercepting and modifying the version check to suggest there was a new version, didn't notice it pop anything (assuming I did it quick enough that it didn't time out internally). It looks like it tried to make a request to download my fake version, which obviously got a 404. Nothing popped up in the UI/etc, but that makes me think it implements some kind of autoupdate.

From the announcement email I believe they said it is built with Tauri:

- https://tauri.app/

Which appears to have a built in concept of an auto-updater:

- https://tauri.app/v1/guides/distribution/updater/
- https://github.com/tauri-apps/tauri/discussions/2776

And the update json file format seems to match what I'm seeing on this request:

- https://tauri.app/v1/guides/distribution/updater/#static-json-file

Seems there are ways to force it to do an update check:

- https://tauri.app/v1/guides/distribution/updater/#checking-for-updates

---

@bevanjkay Based on the above, I'll add `auto_updates true`

**Review by bevanjkay — APPROVED** [review]

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**